### PR TITLE
Update worker_init_fn test for pytorch 1.9 nightly

### LINKED ISSUE #7396: fix pl_worker_init_fn for pytorch 1.9
## 🐛 Bug

Investigate a test.

### To Reproduce

Run this test:

```
FAILED tests/trainer/test_dataloaders.py::test_missing_worker_init_fn - asser...
= 1 failed, 1422 passed, 263 skipped, 3 xfailed, 2760 warnings in 435.18s (0:07:15) =
```

### Expected behavior

obvious. 

### Environment

PyTorch 1.9 (nightly)

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #7396

We can ignore a test for pytorch 1.9 since it successfully implements the seeding for numpy in DataLoader workers. 
Here is where it got implemented:
https://github.com/pytorch/pytorch/commit/3b977a0d2834d300c0301a0c6af98c8e939019ce

Thanks @ananthsub for pointing me to the commit.

cc @Borda 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]
Nice that your initial commit was a reference for the PyTorch fix!

File: tests/trainer/test_dataloaders.py [hunk 1]
```diff
@@ -803,8 +803,12 @@ def _user_worker_init_fn(_):
     pass
 
 
+@RunIf(max_torch="1.8.9")
```
Comment (Borda) [inline tests/trainer/test_dataloaders.py:806]: Shall we rather have an arg that spec max wersion as < instead of <=

**Review by Borda — APPROVED** [review]

File: tests/trainer/test_dataloaders.py (hunk 1 above)
Comment (Borda) [reply tests/trainer/test_dataloaders.py:806]: Or set it just as 1.8 as max without minor numbers?

Comment (awaelchli) [reply tests/trainer/test_dataloaders.py:806]: < would be more practical I guess but it would not be obvious from the name if < or <=. 
Maybe a different approach would be to have torch_gt, torch_ge, torch_lt, torch_le but that would be more argumens ...
not sure

Comment (Borda) [reply tests/trainer/test_dataloaders.py:806]: I may say that more practical would be setting level
- max 1.8.1 will keep 1.8.0, 1.8.1 and ignore 1.8.2 and above
- max 1.8 will keep 1.8.0, 1.8.1, 1.8.x and ignore 1.9.x and above
- max 1 will keep 1.8.0, 1.x and ignore 2.x and above

**Review by justusschock — APPROVED** [review]

File: tests/trainer/test_dataloaders.py (hunk 1 above)
Comment (justusschock) [reply tests/trainer/test_dataloaders.py:806]: +1 for the gt, ge, lt and le.  Another option would be to have something like `torch_req='<1.9``. But this would involve more parsing logic.

Comment (ananthsub) [reply tests/trainer/test_dataloaders.py:806]: @awaelchli should the pl worker init function when used with PT 1.9 also be a no-op? given that pytorch now sets the seed correctly?

Comment (awaelchli) [reply tests/trainer/test_dataloaders.py:806]: for numpy yes, we don't need to worry about it in 1.9+. 

But another part of the `pl_worker_init_fn` is that it derives not only a unique seed for each worker but it makes it also unique across all distributed processes (we do that by incorporating the global_rank into the seed sequence). I don't think 1.9 does that.
